src/engine/TransitivePath.cpp

```diff
@@ -658,119 +449,158 @@ void TransitivePath::computeTransitivePathRightBound(
     }
-
-    last_result_end = res.size();
   }
-
-  *dynRes = std::move(res).toDynamic();
+  return hull;
 }
 
 // _____________________________________________________________________________
-ResultTable TransitivePath::computeResult() {
-  LOG(DEBUG) << "TransitivePath result computation..." << std::endl;
-  shared_ptr<const ResultTable> subRes = _subtree->getResult();
-  LOG(DEBUG) << "TransitivePath subresult computation done." << std::endl;
+template <size_t WIDTH, size_t TEMP_WIDTH>
+void TransitivePath::fillTableWithHull(IdTableStatic<WIDTH>& table, Map hull,
+                                       std::vector<Id>& nodes,
+                                       size_t startSideCol,
+                                       size_t targetSideCol,
+                                       const IdTable& tableTemplate,
+                                       size_t skipCol) {
+  IdTableView<TEMP_WIDTH> templateView =
+      tableTemplate.asStaticView<TEMP_WIDTH>();
+
+  size_t rowIndex = 0;
+  for (size_t i = 0; i < nodes.size(); i++) {
+    Id node = nodes[i];
+    if (!hull.contains(node)) {
+      continue;
+    }
 
-  IdTable idTable{getExecutionContext()->getAllocator()};
+    for (Id otherNode : *hull[node]) {
+      table.emplace_back();
+      table(rowIndex, startSideCol) = node;
+      table(rowIndex, targetSideCol) = otherNode;
 
-  idTable.setNumColumns(getResultWidth());
+      TransitivePath::copyColumns<TEMP_WIDTH, WIDTH>(templateView, table, i,
+                                                     rowIndex, skipCol);
 
-  size_t subWidth = subRes->idTable().numColumns();
-  if (_leftSideTree != nullptr) {
-    shared_ptr<const ResultTable> leftRes = _leftSideTree->getResult();
-    size_t leftWidth = leftRes->idTable().numColumns();
-    CALL_FIXED_SIZE((std::array{subWidth, leftWidth, _resultWidth}),
-                    &TransitivePath::computeTransitivePathLeftBound, this,
-                    &idTable, subRes->idTable(), leftRes->idTable(),
-                    _leftSideCol, _rightIsVar, _leftSubCol, _rightSubCol,
-                    _rightValue, _minDist, _maxDist, _resultWidth);
-  } else if (_rightSideTree != nullptr) {
-    shared_ptr<const ResultTable> rightRes = _rightSideTree->getResult();
-    size_t rightWidth = rightRes->idTable().numColumns();
-    CALL_FIXED_SIZE((std::array{subWidth, rightWidth, _resultWidth}),
-                    &TransitivePath::computeTransitivePathRightBound, this,
-                    &idTable, subRes->idTable(), rightRes->idTable(),
-                    _rightSideCol, _leftIsVar, _leftSubCol, _rightSubCol,
-                    _leftValue, _minDist, _maxDist, _resultWidth);
-  } else {
-    CALL_FIXED_SIZE(subWidth, &TransitivePath::computeTransitivePath, this,
-                    &idTable, subRes->idTable(), _leftIsVar, _rightIsVar,
-                    _leftSubCol, _rightSubCol, _leftValue, _rightValue,
-                    _minDist, _maxDist);
+      rowIndex++;
+    }
```

Comment:
This is not quite efficient anymore with the column-based implementation of the IdTables, but, this should probably be in a different PR after setting up a benchmark for which part takes how much time etc.